### DISCUSSION TITLE
use zig or msys2 compiler toolchain (gcc/clang) for windows?

### DISCUSSION THREAD

**Discussion by goyalyashpal** [c3lang/c3c#1146] [Q&A] [answered]
hi! windows dist of c3 requires the horrid msvc. 

src: 
* [install instructions]

https://github.com/c3lang/c3c/blob/9a114b38d3e5d659b69d6a2608c184583dc2c20c/README.md?plain=1#L195-L199

[install instructions]: https://github.com/c3lang/c3c#installing-on-windows-with-precompiled-binaries

it would be nice if it could use the msys2-gcc / msys2-clang toolchain (and optiinally distributed there too).

yes, i have searched with keywords {msys,cygwin} on {issues,discussions}

**Comment by lerno** [comment]
It doesn't require that you use MSVC. But it does need to link against the .dlls so the script downloads the required dependencies.

**Reply by goyalyashpal** [reply]
i would try to see for myself if smth regarding that can be done - just as a hobby. if ud like to share some initialising vectors for that, feel free. 😊😃

**Reply by lerno** [reply]
The thing is that requiring the user to install msys or some such would be a HUGE requirement, as opposed to the current which is to just run a script which downloads .lib files. What is the win?

**Reply by goyalyashpal** [reply]
> _"What is the win?"_

many students (i.e. begginers) doing c/c++ devt already have mingw on their system. and likely, many developers & softwares alraedy use msys2 for "build from src" when doing multi platform devt (like for inkscape etc)

**Comment by goyalyashpal** [comment]
actually, i am running on a public wifi & havent got my laptop with me for now. so, is there some link using which i can dowload it on mobile, and transfer it to pc later?

**Reply by lerno** [reply]
Download what? The compiler or the necessary files for compiling for windows?

**Reply by goyalyashpal** [reply]
i have downloaded the c3-windows.zip (~30 MiB)

i have msys2 on my system, but nothing related to visual studio is available there. so, whatever would be required to get c3 up & running on my pc. 

thanks a lot for prompt responses & helping me out.

**Reply by lerno** [reply]
The .py script works on any computer, zip the resulting directory and unpack it in the same directory as c3c.exe. It would be automatically included in C3 if it wasn't for copyright issues.

**Comment by goyalyashpal** [comment]
hey! how about using the `zig` compiler? 

its lightweight, mit license, cross platform, cross compiling, sane defaults, provides caching. can even pick different library (musl, glibc, etc.)

https://github.com/ziglang/zig

**Reply by lerno** [reply]
For doing what exactly?

**Reply by goyalyashpal** [reply]
for doing what `msvc` is currently doing in `c3lang` on windows, & other things elsewhere
i.e. providing those `lib/std/*`, `*_sdk/<arch>/*`

and also u mentioned the following:, which i suppose wont be an issue with zig:

> ... It would be automatically included in C3 if it wasn't for copyright issues.
> \- @ lerno at https://github.com/c3lang/c3c/discussions/1146#discussioncomment-8517082

**Reply by lerno** [reply]
But using the Zig compiler won't help with that. Zig merely uses the Mingw .lib files which are not copyright encumbered, but otherwise correspond to the MSVC .lib files. I do not want to depend on mingw being correct or up to date. Thus the .py script, which at a later point might be integrated into the compiler. The bottom line is that I want perfect integration with MSVC, not merely good.

**Reply by goyalyashpal** [reply]
> The bottom line is that I want perfect integration with MSVC ...

the reason for that being that its (supposed to be) literally _THE stdlib_ talking to the os kernel for windows right?

> .py script ... at a later point might be integrated into the compiler

i dont think that would be any good.

**Reply by lerno** [reply]
> the reason for that being that its (supposed to be) literally THE stdlib talking to the os kernel for windows right?

I don't know what you mean by this.

Also, I don't know why you mind having the compiler download the MSVC sdk on demand.

**Reply by goyalyashpal** [reply]
i meant that "they are supposed to be the perfect stdlib for windows" - is this the reason u want to use msvc

i dont have a prob with msvc. i just want to know the reason

**Reply by lerno** [reply, marked as answer]
I don't want people to run into any issues linking libraries compiled with MSVC.
